folio-spring-base/src/test/java/org/folio/spring/client/EnrichUrlAndHeadersClientTest.java

```diff
@@ -0,0 +1,189 @@
+package org.folio.spring.client;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.aMapWithSize;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import feign.Request;
+import feign.Response;
+import feign.okhttp.OkHttpClient;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.testing.type.UnitTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.springframework.test.util.ReflectionTestUtils;
+
+@UnitTest
+class EnrichUrlAndHeadersClientTest {
+
+  static List<Arguments> urlPreparationCases() {
+    return Arrays.asList(
+      arguments("http://test-url", null, "http://test-url"),
+      arguments("http://test-url", "http://okapi", "http://okapi/test-url"),
+      arguments("http://test-url", "http://okapi/", "http://okapi/test-url")
+    );
```

Comment:
Nit: I think `List.of()` is more semantically appropriate
```suggestion
    return List.of(
      arguments("http://test-url", null, "http://test-url"),
      arguments("http://test-url", "http://okapi", "http://okapi/test-url"),
      arguments("http://test-url", "http://okapi/", "http://okapi/test-url")
    );
```